google-cloud-graalvm-samples/google-cloud-graalvm-pubsub-sample/src/main/java/com/example/PubSubSampleApplication.java

```diff
@@ -17,35 +17,78 @@
 package com.example;
 
 import com.google.api.core.ApiFuture;
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.cloud.ServiceOptions;
 import com.google.cloud.pubsub.v1.Publisher;
+import com.google.cloud.pubsub.v1.SubscriptionAdminClient;
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.cloud.pubsub.v1.stub.GrpcSubscriberStub;
+import com.google.cloud.pubsub.v1.stub.SubscriberStub;
+import com.google.cloud.pubsub.v1.stub.SubscriberStubSettings;
 import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.AcknowledgeRequest;
+import com.google.pubsub.v1.ProjectSubscriptionName;
 import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.PullRequest;
+import com.google.pubsub.v1.PullResponse;
+import com.google.pubsub.v1.PushConfig;
+import com.google.pubsub.v1.ReceivedMessage;
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.Topic;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
 
 public class PubSubSampleApplication {
 
+  private static final String PROJECT_ID = ServiceOptions.getDefaultProjectId();
+
   /**
    * Driver for the Pub/Sub Sample application which publishes a message to a specified topic.
-   *
-   * <p>The topic should be specified over command line in the form:
-   * java -jar YOUR_JAR.jar projects/YOUR_PROJECT_ID/topics/YOUR_TOPIC_NAME
    */
   public static void main(String[] args) throws Exception {
-    if (args.length <= 0) {
-      System.err.println("Error: Please provided a Pub/Sub topic name as an additional argument "
-          + "to invoking this JAR. The command should be in the form: "
-          + "java -jar YOUR_JAR.jar projects/YOUR_PROJECT_ID/topics/YOUR_TOPIC_NAME");
+    String topicId = "graal-pubsub-test-" + UUID.randomUUID().toString();
+    String subscriptionId = "graal-pubsub-test-sub" + UUID.randomUUID().toString();
+
+    createTopic(topicId);
+    createSubscription(subscriptionId, topicId);
+
+    publishMessage(topicId);
+    subscribeSync(subscriptionId);
+
+    deleteTopic(topicId);
```

Comment:
Might want to do cleanup in try/finally, so if there is an exception, randomly named resources are cleaned up.

Reply:
Good call, done.